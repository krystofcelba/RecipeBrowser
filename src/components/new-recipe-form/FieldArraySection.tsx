import React from 'react';
import { Section } from 'react-native-tableview-simple';
import { GenericFieldArray } from 'redux-form';

import i18n from 'src/assets/i18n';
import ErrorsSectionFooter from './ErrorsSectionFooter';
import ButtonCell from './ButtonCell';

interface Props {
  header?: string;
  fields: GenericFieldArray;
  fieldComponent: React.SFC<any> | React.ComponentClass<any> | string;
  errors?: [string];
  testIDcontext: string;
}

const FieldArraySection = ({ fields, errors = {}, header, fieldComponent, testIDcontext, ...otherProps }: Props) => (
  <Section header={header} footerComponent={<ErrorsSectionFooter testIDcontext={testIDcontext} errors={errors} />}>
    {fields.map((field, i) => React.createElement(fieldComponent, { ...otherProps, key: i, index: i, name: field }))}
    <ButtonCell testID={`${testIDcontext}:add`} title={`+ ${i18n.t('add')}`} onPress={fields.push} />
  </Section>
);

export default FieldArraySection;
