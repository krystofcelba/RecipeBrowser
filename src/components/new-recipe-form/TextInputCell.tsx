import React from 'react';
import { Cell } from 'react-native-tableview-simple';
import { Field } from 'redux-form/immutable';

import { View, BoldText } from 'src/components/common';
import AppTheme from 'src/assets/appTheme';

import FormTextInput from './FormTextInput';

interface Props {
  index?: number;
  name: string;
  placeholder?: string;
  numberOfLines?: number;
}

const TextInputCell = ({ index, ...otherProps }: Props) => {
  const showIndex = index !== undefined;
  return (
    <Cell
      cellContentView={
        <View flexDirection="row" flex={1} paddingVertical={AppTheme.defaultPadding / 2}>
          {showIndex && <BoldText>#{index + 1}</BoldText>}
          <Field
            component={FormTextInput}
            {...otherProps}
            style={showIndex && { paddingLeft: AppTheme.defaultPadding }}
          />
        </View>
      }
    />
  );
};

export default TextInputCell;
