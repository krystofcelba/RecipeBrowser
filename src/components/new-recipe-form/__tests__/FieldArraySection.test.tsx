import React from 'react';
import { shallow } from 'enzyme';

import FieldArraySection from '../FieldArraySection';
import TextInputCell from '../TextInputCell';

describe('FieldArraySection component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<FieldArraySection fields={[]} header="header" fieldComponent={TextInputCell} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with fields', () => {
    const wrapper = shallow(<FieldArraySection placeholder="placeholder" fields={['field[0]']} header="header" fieldComponent={TextInputCell} />);
    expect(wrapper).toMatchSnapshot();
  });
});
