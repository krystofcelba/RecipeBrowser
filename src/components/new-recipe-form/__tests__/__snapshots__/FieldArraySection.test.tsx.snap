// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldArraySection component renders as expected 1`] = `
<Section
  allowFontScaling={true}
  footer={null}
  footerComponent={
    <ErrorsSectionFooter
      errors={Object {}}
      testIDcontext={undefined}
    />
  }
  footerTextColor="#6d6d72"
  header="header"
  headerComponent={null}
  headerTextColor="#6D6D72"
  hideSeparator={false}
  sectionPaddingBottom={15}
  sectionPaddingTop={15}
  sectionTintColor="#EFEFF4"
  separatorInsetLeft={15}
  separatorInsetRight={0}
  separatorTintColor="#C8C7CC"
>
  <ButtonCell
    onPress={[Function]}
    testID="undefined:add"
    title="+ Add"
  />
</Section>
`;

exports[`FieldArraySection component renders as expected with fields 1`] = `
<Section
  allowFontScaling={true}
  footer={null}
  footerComponent={
    <ErrorsSectionFooter
      errors={Object {}}
      testIDcontext={undefined}
    />
  }
  footerTextColor="#6d6d72"
  header="header"
  headerComponent={null}
  headerTextColor="#6D6D72"
  hideSeparator={false}
  sectionPaddingBottom={15}
  sectionPaddingTop={15}
  sectionTintColor="#EFEFF4"
  separatorInsetLeft={15}
  separatorInsetRight={0}
  separatorTintColor="#C8C7CC"
>
  <TextInputCell
    index={0}
    key="0"
    name="field[0]"
    placeholder="placeholder"
  />
  <ButtonCell
    onPress={[Function]}
    testID="undefined:add"
    title="+ Add"
  />
</Section>
`;
